### ISSUE TITLE
Photos in main menu not working

### ISSUE THREAD

**Issue by timmersr** [nextcloud/android#1586]
### Actual behaviour
On selection of Photos from the menu "Loading..." is displayed

### Expected behaviour
Display photos.
 
### Steps to reproduce
1. select Photos from the menu.
2. 
3. 

### Environment data
Android version: 7.0
Device model: Samsung Galaxy 6 Edge Plus
Stock or customized system: Stock
Nextcloud app version: 2.0.0RC7
Nextcloud server version: 12.0.3

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by alexknobbe** [comment]
Same on my side.
Android 7
Huawai P8 lite

**Comment by mario** [comment]
@alexknobbe @timmersr logs? :)

**Comment by e-alfred** [comment]
@mario I have the same problem with 2.0 RC6. The loading screen is shown indefinitely. I see nothing in the server logs.

**Comment by mario** [comment]
Try rc7

**Comment by e-alfred** [comment]
Will do as soon as it is available in F-Droid.

**Comment by timmersr** [comment]
@mario Nothing in the server logs.

**Comment by alexknobbe** [comment]
Nothing in logs here and using RC7

**Comment by j-ed** [comment]
Android 7.1.2
Nextcloud app 2.0.0 RC7

I can confirm the described problem on my phone with the latest build. I must admit that the initial load of the Gallery app on my desktops also lasts around 80s, but on my phone nothing is shown although I've waited several minutes for a reply. Accessing image files using the normal directory pathes nearly show all files immediately.

**Comment by AndyScherzinger** [comment]
If you still see the loading spinner then it is likely that you need to wait even longer.
I for example have to wait for 12 minutes. This will be addressed in the post v2.0.0 era.
The reason is rather simple. The client still uses the so called list view which means the content will be completely rendered in one go which depending on the amount of files kills performance incredibly hard. So the render issue will be addressed but that leads to a complete rewrite of the UI of all file listings since they share the same implementation.
The second aspect would then be to rather have some kind of block reading instead of fetching thousands of results.

**Comment by timmersr** [comment]
IMHO options in the main menu that (effectively) do not work, should be removed until they do.

**Comment by e-alfred** [comment]
I think @timmersr is right. Having a "feature" that loads for probably tens of minutes just because there are a lot of pictures on the server is just useless. Who would want to wait for that long just to view all pictures on a mobile device (!) if they are in a hurry or just would like to have a gallery view of all of their images?

**Comment by j-ed** [comment]
@AndyScherzinger Thank you for this insights, that explains the observed behavior. I would also agree to temporarily remove the menu entry until the underlying code base works as expected because no-one waits minutes to get anything displayed. Especially not if the other ways are working reliable.

**Comment by AndyScherzinger** [comment]
> Especially not if the other ways are working reliable.

They don't ;) To mimic this in the normal list view you just need to have a folder with 5000 images (yes there have been reports of users with such high volume folders...). About hiding these entries, I'd leave this decision to @tobiasKaminsky and @mario. Depending on the timeline when we could implement/publish the recyclerview implementation.

**Comment by tobiasKaminsky** [comment]
#1901 might be a nice way to fix it.

**Comment by tobiasKaminsky** [comment]
Technically it is working. In reality it is way to slow if many photos are involved.
There is already another issue and also a PR trying to fix it.

It also depends on server to enhance webdav search.

--> closing this bug, but let the enhancement / pr open for further improvement.

**Comment by rcdailey** [comment]
@tobiasKaminsky Recommend we reopen until #1901 is merged. I personally feel closing issues prior to a resolution being merged is harmful since the issue loses visibility.

**Comment by tobiasKaminsky** [comment]
#1901 is no Pull Request but a way how to solve this problem.
I think it is fine to keep this closed, but #1901 open.
